### PR TITLE
onesync: Add GetVehiclePedIsIn

### PR DESCRIPTION
**Pull request by Quattro20v**
My attempt of adding in GetVehiclePedIsIn to the server-side. I think that it would be highly useful to double check what vehicle the player is in on the server-side.

Don't bash me too much tho it's my first time looking at this source code. Plus I am not even sure how to test this unfortunately and run it. And for some reason "citizen-server-impl" didn't show up in the CitizenMP.sln, so I had to manually open the whole folder.

Either way, let me know how does this look, if it makes sense and maybe some tips on how to contribute and test the code in the future.

### CODE REVIEW

**Comment by PichotM** [comment]
lastVehicle is only in the struct definition and is never set so it won't work
but the rest should be fine, thanks for this PR

**Comment by PichotM** [comment]
And also use MakeHandle
As I believe it returns the netid

**Comment by Quattro20v** [comment]
Thank you for taking a look! I just checked the lastVehicle, are you sure that lastVehicle is never set? I found some code in ServerGameState.cpp:
line: 1422 `vehicleData->lastVehicle = vehicleData->curVehicle;` 

Also by MakeHandle do you mean MakeScriptHandle? That's what "GET_ALL_VEHICLES" seems to use.

**Comment by Quattro20v** [comment]
Seems like I messed it up, with trying to do the MakeScriptHandle. Maybe someone can shine some light in here.

Also still waiting for some clarification if lastVehicle exists or not.

**Comment by Quattro20v** [comment]
After figuring out how to run the server and test, I got it to finally work. Most of the proper work is in the last 2 commits. I pulled the trigger too soon with the other ones, before getting familiar with the source code and how to test it.

File: code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp [hunk 1]
```diff
@@ -802,6 +802,40 @@ static InitFunction initFunction([]()
...
+		if ((lastVehicleArg == true && node->lastVehiclePedWasIn == -1) || (lastVehicleArg == false && node->curVehicle == -1))
+			return (uint32_t)0;
+
+		auto returnEntity = lastVehicleArg == true ? gameState->GetEntity(0, node->lastVehiclePedWasIn) : gameState->GetEntity(0, node->curVehicle);
+
+		if (!returnEntity)
+			return (uint32_t)0;
+
+		// Return current or last vehicle
+		if (lastVehicleArg)
```
Comment (jabbink) [inline code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp:834]: 4 lines to return `gameState->MakeScriptHandle(returnEntity);`

Comment (Quattro20v) [reply code/components/citizen-server-impl/src/state/ServerGameState_Scripting.cpp]: Good catch! Fixed.

**Comment by blattersturm** [comment]
Can you maybe rebase this onto latest master and add a native declaration to ext/native-decls?

**Comment by Quattro20v** [comment]
Yeah for sure, I can do that when I wake up!

**Comment by Quattro20v** [comment]
Native citizenfx/natives#289

**Comment by blattersturm** [comment]
> Native [citizenfx/natives#289](https://github.com/citizenfx/natives/pull/289)

no, natives go in ext/native-decls, not the natives repo

**Comment by Quattro20v** [comment]
Added the native declaration to ext/native-decls

**Comment by blattersturm** [comment]
`.MD`->`.md`, perhaps; filenames are case-sensitive on some systems.

**Comment by Quattro20v** [comment]
Fixed the file name extension. Sorry about that.

**Comment by blattersturm** [comment]
OK, since Travis succeeded (and Appveyor config was broken), this might get merged in soon. :)

**Comment by Quattro20v** [comment]
Sounds good, thank you :)

**Pull request merged**